db_connection.py

```diff
@@ -49,24 +51,89 @@ def start_docker_client(self):
 
     def collect_stats(self):
         containers = self.docker_client.containers.list()
+        if len(containers) == 0:
+            return
+        
         for container in containers:
             stats = container.stats(stream=False)
-            cpu_usage = stats['cpu_stats']['cpu_usage']['total_usage']
 
-            if not 'memory_stats' in stats:
-                memory_usage = None
-            else:
-                memory_usage = stats['memory_stats']['usage']
+            container_name = stats['name']
+            container_id = stats['id']
+
+            memory_stats = self.get_container_memory_statistics(stats)
+            cpu_stats = self.get_container_cpu_statistics(stats)
+            network_stats = self.get_container_network_stats(stats)
             
-            container_id = container.id
+            try:
+                self.cursor.execute("SELECT EXISTS(SELECT 1 FROM container_info WHERE id = %s);", (container_id,))
+                container_exists = self.cursor.fetchone()[0]
+                if not container_exists:
+                    self.cursor.execute(
+                        "INSERT INTO container_info (id, name, memory_limit) VALUES (%s, %s, %s)",
+                        (container_id, container_name, memory_stats['memory_limit'])
+                    )
+
+                self.cursor.execute(
+                    "INSERT INTO container_stats (container_id, cpu_usage, memory_usage) VALUES (%s, %s, %s)",
+                    (container_id, cpu_stats['cpu_usage'], memory_stats['memory_usage'])
+                )
+                self.cursor.execute(
+                    "INSERT INTO container_detailed_statistics (id, kernel_cpu_usage, user_cpu_usage, system_cpu_usage) VALUES (%s, %s, %s, %s)",
+                    (container_id, cpu_stats['kernel_cpu_usage'], cpu_stats['user_cpu_usage'], cpu_stats['system_cpu_usage'])
+                )
 
-            self.cursor.execute(
-                "INSERT INTO container_stats (container_id, cpu_usage, memory_usage) VALUES (%s, %s, %s)",
-                (container_id, cpu_usage, memory_usage)
-            )
-            self.conn.commit()
+                for network, net_stat in network_stats.items():
+                    self.cursor.execute(
+                        "INSERT INTO container_networks (id, network_name, resieved_bytes, transmited_bytes) VALUES (%s, %s, %s, %s)",
+                        (container_id, network, net_stat['resieved_bytes'], net_stat['transmited_bytes'])
+                    )
+                self.conn.commit()
+            except errors.UniqueViolation as e:
```

Comment:
What is idea to catch different types of exceptions, including universal `Exception`, but to process them in a same way?

Reply:
Resolved. For now it's useless